link/src/main/java/com/stripe/android/link/LinkActivity.kt

```diff
@@ -19,30 +24,47 @@ import com.stripe.android.link.ui.paymentmenthod.PaymentMethodScreen
 import com.stripe.android.link.ui.signup.SignUpScreen
 import com.stripe.android.link.ui.verification.VerificationScreen
 import com.stripe.android.link.ui.wallet.WalletScreen
+import com.stripe.android.ui.core.CircularProgressIndicator
 
-internal class LinkActivity : AppCompatActivity() {
-    @VisibleForTesting
-    internal var viewModelFactory: ViewModelProvider.Factory = LinkActivityViewModel.Factory()
-    private val viewModel: LinkActivityViewModel by viewModels { viewModelFactory }
+internal class LinkActivity : ComponentActivity() {
+    internal var viewModel: LinkActivityViewModel? = null
```

Comment:
can you make this a `lateinit` variable? I think then you won't need to make the type nullable

Reply:
I made it nullable because `onDestroy` will get called when we finish the activity and `viewModel.unregisterActivity()` will crash since it's not initialized

Reply:
ahh gotcha, thanks